Cleanup: Minor package and API update to snapcraft.yaml

Kedstar99 opened the pull request:
These updates were after trying to get native theming support for the snaps to work based on the changes done on the Thunderbird and LibreOffice snaps.

I tried adding a theming fix in data-dir work around as part of https://github.com/snapcore/snapd/pull/5395 and the themes were successfully loaded into the snap.

The issue is that it appears to not work as I don't think vlc uses libgtk-3-0 to be able to use GTK-3 themes. I tried various environment variables suggested by various forums to get QT to use the gtk themes but none of them really worked.

Whilst using the vlc docker to build the snap, certain aspects of the snap appear to be outdated. For example, prepare, build and install has now been deprecated in trade of override-build.

That and vlc snap appeared to be pulling pacakges from the container which may soon be deprecated too.

If the snap interface theming support evolves, I will try again then.

thresheek commented:
thanks for the contribution @Kedstar99.

I don't see the need to include libqt5x11extras5 to the stage-packages, are you sure that is needed?  That doesnt happen on our builders:

```
Preparing to build wrapper 
Building wrapper 
Staging vlc 
Staging wrapper 
Priming desktop-qt5-vlc 
Priming vlc 
Files from the build host were migrated into the snap to satisfy dependencies that would otherwise not be met. This feature will be removed in a future release. If these libraries are needed in the final snap, ensure that the following are either satisfied by a stage-packages entry or through a part:
usr/lib/x86_64-linux-gnu/libdb-5.3.so
Priming wrapper 
Determining the version from the project repo (version-script).
The version has been set to '3.0.4-141-gc575cef'
snapcraft snap
```

Kedstar99 commented:
Sorry, 
I removed the docker image and build repo from my machine as they take up quite a lot of space to build. As such, I can't remember why I added that dependency. I think it's either was suggested for the desktop-qt5 plug or it may have been left over from some of my prototyping trying to remove the breeze dependencies to see if it could use the gtk3 plugs. Sorry about that.

If you want, I can update my pull request and remove that dependency.

thresheek commented:
Yes, please do.

On Tue, Oct 23, 2018, 18:40 Kedstar99 <notifications@github.com> wrote:

> Sorry,
> I removed the docker image and build repo from my machine as they take up
> quite a lot of space to build. As such, I can't remember why I added that
> dependency. I think it's either was suggested for the desktop-qt5 plug or
> it may have been left over from some of my prototyping trying to remove the
> breeze dependencies to see if it could use the gtk3 plugs. Sorry about that.
>
> If you want, I can update my pull request and remove that dependency.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/videolan/vlc/pull/75#issuecomment-432296400>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AE3xLyzKSQlLzZxBxjqhWXy03QsidxaDks5unzhogaJpZM4XWfTc>
> .
>

Kedstar99 commented:
Done

thresheek commented:
Merged in 22ed204fbd3f7ec33a235c580c4909966a757ad2, thanks!
